dev/src/glsystem.cpp

```diff
@@ -72,6 +73,34 @@ void ClearFrameBuffer(const float3 &c) {
     GL_CALL(glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT));
 }
 
+void SetScissorRect(int2 topleft, int2 size, int2& prev_topleft, int2& prev_size) {
+    int2 scrnsz = GetScreenSize();
+    GLboolean enabled;
+
+    GL_CALL(glGetBooleanv(GL_SCISSOR_TEST, &enabled));
+    if (enabled) {
+        GLint curbox[4];
```

Comment:
make this function take a `pair<int2, int2> &prev` instead, then you can call `glGetBooleanv(GL_SCISSOR_TEST, prev.first.data())` and avoid all the copying. And below you can then `prev = { int2_0, scrnsz }`